indexing-service/src/main/java/io/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -234,50 +265,60 @@ public TaskLock lock(final Task task, final Interval interval) throws Interrupte
       final TaskLockPosse posseToUse;
 
       if (foundPosses.size() > 1) {
-
-        // Too many existing locks.
-        return Optional.absent();
+        Pair<Boolean, Map<String, TaskLock>> pair = canAcquireLock(foundPosses, priority);
+        if (pair.lhs) {
+          revokeTaskLocks(pair.rhs);
```

Comment:
For a posse with multiple tasks, this will call revoke on the same TaskLock multiple times. Is that ok?